django/http/request.py

```diff
@@ -230,6 +288,17 @@ def body(self):
         if not hasattr(self, '_body'):
             if self._read_started:
                 raise RawPostDataException("You cannot access body after reading from request's data stream")
+
+            # Limit the maximum request data size that will be handled in-memory.
+            try:
+                content_length = int(self.META.get('HTTP_CONTENT_LENGTH', self.META.get('CONTENT_LENGTH', 0)))
+            except (ValueError, TypeError):
+                content_length = 0
+
+            if (settings.DATA_UPLOAD_MAX_MEMORY_SIZE is not None
```

Comment:
This check is done for all mime-types, but should be skipped for "multipart/form-data".